userspace/libscap/scap_open.h

```diff
@@ -100,6 +101,8 @@ typedef struct scap_open_args
 
 	scap_source_plugin* input_plugin; ///< use this to configure a source plugin that will produce the events for this capture
 	char* input_plugin_params; ///< optional parameters string for the source plugin pointed by src_plugin
+
+	scap_test_input_data* test_input_data; ///< only used for testing scap consumers by supplying arbitrary test data
```

Comment:
Probably this is not the right PR to change it, but we need to introduce an `enum` to understand which engine we are going to use...

Reply:
If only we had tagged unions in C... :(

If/when we touch this struct, ideally I'd make it contain a field to identify the engine somehow (yeah, probably an enum), some common stuff (callbacks, suppressed comms etc.) and a single `void*` pointing to whatever the engine wants it to point

Reply:
Just one question @gnosek, are you planning to modify it in your vtable PRs or do we want to address it separately?

Reply:
I think it's better to do it after my open path cleanups so I can do that too.